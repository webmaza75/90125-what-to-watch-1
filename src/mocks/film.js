const film = {
  genre: [`Drama`, `Thriller`],
  title: `Aviator`,
  desc: ``,
  picture: `aviator.jpg`,
  year: 2016,
  src: ``
};

export default film;
