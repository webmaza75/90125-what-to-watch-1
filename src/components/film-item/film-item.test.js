import React from 'react';
import renderer from 'react-test-renderer';

import {MemoryRouter} from 'react-router-dom';
import FilmItem from './film-item.jsx';
import film from '../../mocks/film.js';

window.HTMLMediaElement.prototype.play = () => {};

describe(``, () => {
  it(`Film item correctly renders with isPlaying false`, () => {
    const tree = renderer
      .create(<MemoryRouter initialEntries={[`/`]}>
        <FilmItem
          item={film}
          isPlaying={false}
          onMouseOver={jest.fn()}
          onMouseOut={jest.fn()}
        />
      </MemoryRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Film item correctly renders with isPlaying true`, () => {
    const tree = renderer
      .create(<MemoryRouter initialEntries={[`/`]}>
        <FilmItem
          item={film}
          isPlaying={true}
          onMouseOver={jest.fn()}
          onMouseOut={jest.fn()}
          isNeedReload={false}
        />
      </MemoryRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
