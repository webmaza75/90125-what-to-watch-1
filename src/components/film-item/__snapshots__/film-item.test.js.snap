// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film item correctly renders with isPlaying false 1`] = `
<article
  className="small-movie-card catalog__movies-card"
>
  <a
    href="/film/1"
    onClick={[Function]}
  >
    <div
      className="small-movie-card__image"
      onMouseOut={[MockFunction]}
      onMouseOver={[Function]}
    >
      <video
        className="player__video"
        muted={true}
        poster="https://es31-server.appspot.com/wtw/static/film/preview/war-of-the-worlds.jpg"
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
      />
    </div>
  </a>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/film/1"
      onClick={[Function]}
    >
      War of the Worlds
    </a>
  </h3>
</article>
`;

exports[`Film item correctly renders with isPlaying true 1`] = `
<article
  className="small-movie-card catalog__movies-card"
>
  <a
    href="/film/1"
    onClick={[Function]}
  >
    <div
      className="small-movie-card__image"
      onMouseOut={[MockFunction]}
      onMouseOver={[Function]}
    >
      <video
        className="player__video"
        muted={true}
        poster="https://es31-server.appspot.com/wtw/static/film/preview/war-of-the-worlds.jpg"
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
      />
    </div>
  </a>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/film/1"
      onClick={[Function]}
    >
      War of the Worlds
    </a>
  </h3>
</article>
`;
